src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,795 @@
+                    if (response.WorkOnRelease || response.WaitForPartition)
+                    {
+                        timeToSleep = TimeSpan.FromSeconds(1);
+                    }
+                }
+                catch(RequestFailedException ex) when (ex.Status == 412)
+                {
+                    // if the worker failed to update the table, re-read the table immediately without waiting.
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+                catch(Exception exception)
+                {
+                    this.settings.Logger.PartitionManagerError(
+                       this.storageAccountName,
+                       this.settings.TaskHubName,
+                       this.settings.WorkerId,
+                       "",
+                       $"Errors in partition manager reading the table : {exception}");
+                }
+                await Task.Delay(timeToSleep, token);
+            }
+        }
+
+        /// <summary>
+        /// This method will stop the partition manager. It first stops the task ReadAndWriteTable().
+        /// And then start the Task ShutDown() until all the leases in the worker is drained. 
+        /// </summary>
+        async Task IPartitionManager.StopAsync()
+        {
+            this.partitionManagerCancellationSource.Cancel();
+            
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",
+                $"Starts draining the in-memory messages of all ownership leases for shutdown.");
+
+            bool isFinish = false;
+            //Shutting down is to drain all the current leases and then release them.
+            //The worker checks table every 1 second to see if the realease of all ownership lease finishes to ensure timely updates.
+            var timeToSleep = TimeSpan.FromSeconds(1);
+
+            //Waiting for finish draining all the ownership leases and release them. 
+            while (!isFinish)
+            {
+                try
+                {
+                    isFinish = await this.tableLeaseManager.ShutDown();
+                }
+                //If the worker failed to update the table due to concurrency, re-read the table immediately without wait.
+                catch (RequestFailedException ex) when (ex.Status == 412)
+                {
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+                //Eat any other exceptions.
+                catch(Exception exception)
+                {
+                    this.settings.Logger.PartitionManagerError(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        "",
+                        $"Error in stopping partition manager : {exception}");
```

Comment:
We should add a note in the message indicating that we will continue retrying.
```suggestion
                        $"Error in stopping partition manager. Will continue retrying. Error details: {exception}");
```